dpypelines/pipeline/utils.py

```diff
@@ -363,56 +362,136 @@ def get_distribution_details_for_request(dcat_distributions: list) -> list:
         ) from err
 
 
-def upload_metadata(local_store, email_client, submitter_email):
-    """Upload metadata and send notifications."""
+def check_dataset_type_is_static(dataset_api_client: DatasetAPIClient) -> bool:
+    get_dataset_result = dataset_api_client.get(
+        f"{dataset_api_client.dataset_api_url}/{dataset_api_client.dataset_path}",
+        headers=dataset_api_client.token_auth.get_auth_header(),
+    )
+    if get_dataset_result.status_code == 200:
```

Comment:
I'd recommend swapping the if cases here. I.e.
```suggestion
    if get_dataset_result.status_code != 200:
      get_dataset_result.raise_for_status()
      
    <rest of code>
```

Generally always try and exit early if possible, as you'll reduce nesting and make the code cleaner/easier to read :)